timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -0,0 +1,366 @@
+        minimum_resources = {"num_cpus": 1}
+        # if GPU is available, we train with 1 GPU per trial
+        if is_gpu_available:
+            minimum_resources["num_gpus"] = MODEL_CONFIGS[self.model_path].get("num_gpus", 0)
+        return minimum_resources
+
+    def _get_model_pipeline(self):
+        gpu_available = self._is_gpu_available()
+
+        assert self.model_path in MODEL_CONFIGS or Path.is_dir(
+            Path(self.model_path)
+        ), f"Model path {self.model_path} is not supported"
+
+        if not gpu_available and MODEL_CONFIGS.get(self.model_path, {}).get("num_gpus", 0) > 0:
+            raise RuntimeError(
+                f"{self.name} requires a GPU to run, but no GPU was detected. "
+                "Please make sure that you are using a GPU instance and "
+                "`import torch; torch.cuda.is_available()` returns `True`."
+            )
+
+        device = self.device or ("cuda" if gpu_available else "auto")
+
+        pipeline = OptimizedChronosPipeline.from_pretrained(
+            self.model_path,
+            device_map=device,
+            optimization_strategy=self.optimization_strategy,
+        )
+
+        return pipeline
+
+    def _fit(
+        self,
+        train_data: TimeSeriesDataFrame,
+        val_data: Optional[TimeSeriesDataFrame] = None,
+        time_limit: int = None,
+        **kwargs,
+    ) -> None:
+        self._check_fit_params()
+        self.model_pipeline = self._get_model_pipeline()
+
+    def get_inference_data_loader(
+        self,
+        data: TimeSeriesDataFrame,
+        num_workers: int = 1,
+    ):
+        import torch
+
+        chronos_dataset = ChronosInferenceDataset(
+            target_df=data,
+            target_column=self.target,
+            context_length=self.context_length,
+        )
+
+        return torch.utils.data.DataLoader(
+            chronos_dataset,
+            batch_size=self.batch_size,
+            shuffle=False,
+            num_workers=num_workers,
+        )
+
+    def _predict(
+        self,
+        data: TimeSeriesDataFrame,
+        known_covariates: Optional[TimeSeriesDataFrame] = None,
+        **kwargs,
+    ) -> TimeSeriesDataFrame:
+        import torch
+
+        if self.model_pipeline is None:
+            raise ValueError("Please fit the model before predicting.")
+
+        self.model_pipeline.model.eval()
+        with torch.no_grad():
+            prediction_samples = (
+                self.model_pipeline.predict(
+                    batch,
+                    prediction_length=self.prediction_length,
+                    num_samples=self.num_samples,
+                    limit_prediction_length=False,
+                )
+                .detach()
+                .cpu()
+                .numpy()
+                for batch in self.get_inference_data_loader(data=data)
+            )
+
+        samples = np.concatenate([c.T for c in chain.from_iterable(prediction_samples)], axis=0)
```

Comment:
If each entry in `prediction_samples` is an array of shape `[batch_size, prediction_length, num_samples]`, could we get the same result with `np.concatenate(prediction_samples, axis=0)`?

Reply:
`chain.from_iterable` processes item by item in each batch, thus reshaping as well. The alternative is something along the lines of 
```
X2 = np.concatenate(list(pred_samples), axis=0)
res2 = X2.swapaxes(0, 1).reshape(X2.shape[1], -1)
```

for which I did not see a consistent speedup.